contracts/contracts/buyback/BaseBuyback.sol

```diff
@@ -135,110 +153,135 @@ abstract contract BaseBuyback is Initializable, Strategizable {
     }
 
     /**
-     * @dev Swaps half of `oTokenAmount` to OGV
-     *      and the rest to CVX and finally lock up CVX
-     * @param oTokenAmount Amount of OUSD/OETH to swap
-     * @param minOGV Minimum OGV to receive for oTokenAmount/2
-     * @param minCVX Minimum CVX to receive for oTokenAmount/2
+     * @dev Computes the split of oToken balance that can be
+     *      used for OGV and CVX buybacks.
      */
-    function swap(
+    function _updateBuybackSplits()
+        internal
+        returns (uint256 _ogvShare, uint256 _cvxShare)
+    {
+        _ogvShare = ogvShare;
+        _cvxShare = cvxShare;
+
+        uint256 totalBalance = IERC20(oToken).balanceOf(address(this));
+        uint256 unsplitBalance = totalBalance - _ogvShare - _cvxShare;
+
+        // Check if all balance is accounted for
+        if (unsplitBalance > 0) {
+            // If not, split unaccounted balance 50:50
+            uint256 halfBalance = unsplitBalance / 2;
+            _cvxShare = _cvxShare + halfBalance;
+            _ogvShare = _ogvShare + unsplitBalance - halfBalance;
+
+            // Update storage
+            ogvShare = _ogvShare;
+            cvxShare = _cvxShare;
+        }
+    }
+
+    function updateBuybackSplits() external onlyGovernor {
+        // slither-disable-next-line unused-return
+        _updateBuybackSplits();
+    }
+
+    function _swapToken(
+        address tokenOut,
         uint256 oTokenAmount,
-        uint256 minOGV,
-        uint256 minCVX
-    ) external onlyGovernorOrStrategist nonReentrant {
+        uint256 minAmountOut,
+        bytes calldata swapData
+    ) internal returns (uint256 amountOut) {
         require(oTokenAmount > 0, "Invalid Swap Amount");
-        require(universalRouter != address(0), "Uniswap Router not set");
-        require(rewardsSource != address(0), "RewardsSource contract not set");
-        require(minOGV > 0, "Invalid minAmount for OGV");
-        require(minCVX > 0, "Invalid minAmount for CVX");
+        require(swapRouter != address(0), "Swap Router not set");
+        require(minAmountOut > 0, "Invalid minAmount");
 
-        uint256 ogvBalanceBefore = IERC20(ogv).balanceOf(rewardsSource);
-        uint256 cvxBalanceBefore = IERC20(cvx).balanceOf(address(this));
+        // Transfer OToken to Swapper for swapping
+        // slither-disable-next-line unchecked-transfer unused-return
+        IERC20(oToken).transfer(swapRouter, oTokenAmount);
 
-        uint256 amountInForOGV = oTokenAmount / 2;
-        uint256 amountInForCVX = oTokenAmount - amountInForOGV;
+        // Swap
+        amountOut = ISwapper(swapRouter).swap(
```

Comment:
Could you point me to the corresponding doc?

Reply:
The code is here: https://github.com/OriginProtocol/origin-dollar/blob/master/contracts/contracts/swapper/Swapper1InchV5.sol

Deployed contract is here: <https://etherscan.com/address/0xcD0fcF8a31Bc78ec07752e9CCD3960E936D18366>

I don't think we have document the methods outside the code though